core/src/validator.rs

```diff
@@ -169,6 +171,40 @@ impl Validator {
             bank.slots_per_segment(),
         );
 
+        let working_bank = bank_forks.read().unwrap().working_bank();
+        let validator_pubkey = keypair.pubkey();
+        let mut validators_current_stakes: Vec<_>  = working_bank
+            .epoch_vote_accounts(bank.get_epoch_and_slot_index(working_bank.slot()).0)
+            .map(|accounts| {
+                accounts
+                    .iter()
+                    .filter_map(|(pubkey, (stake, account))| {
+                        if let Ok(vote_state) = VoteState::deserialize(&account.data) {
+                            if vote_state.node_pubkey == validator_pubkey { 
+                                Some((*pubkey, *stake))
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        }
+                    })
+                    .collect()
+            })
+            .unwrap();
+        
+        if validators_current_stakes.len() > 0 {
+            for current_stakes in &validators_current_stakes {
+                let (pubkey, stake) = current_stakes.unwrap();
+                if stake > 0 {
+                    set_handler(move || {
+                        // warn!("Error writing file! {:?}", res);
+                        warn!("Validator with pubkey {:?} has to deactivate {:?} stake", pubkey, stake);
```

Comment:
It'd be nice to have a counter here too.  The first Ctrl-C warns, the second Ctrl-C terminates (or perhaps only if you Ctrl-C a second time with a couple seconds)